Controller/Standard/AbstractStandard.php

```diff
@@ -243,8 +243,13 @@ protected function _retrieveQuote()
             $this->_logger->error(__("Session Quote doesnot exist."));
             $use_checkout_api_quote = true;
         } elseif ($checkout_id != $zipMoneyCheckoutId) {
-            $this->_logger->error(__("Checkout Id doesnot match with the session quote."));
-            $use_checkout_api_quote = true;
+            // iframe redirect does not have au prefix for AU stack
+            if ($checkout_id != 'au-'.$zipMoneyCheckoutId) {
+                $this->_logger->error(__("Checkout Id doesnot match with the session quote."));
+                $use_checkout_api_quote = true;
+            } else {
+                return $sessionQuote;
+            }
```

Comment:
Please double check my logic because I'm just guessing, but...

Instead of adding this additional `if/else` statement and adding more nesting which can make it harder to maintain and read, can we change `line 245` to be?:
`elseif ($checkout_id != $zipMoneyCheckoutId && $checkout_id != 'au-'.$zipMoneyCheckoutId) {`

Reply:
I did this but not working. I need nested if condition.

Reply:
I have approved for now, but would be interesting to understand why it didn't work. 🤔

Reply:
Sorry it is working. I have changed the condition.